tools/rtpsrelay/GuidAddrSet.cpp

```diff
@@ -36,18 +36,22 @@ GuidAddrSet::record_activity(const Proxy& proxy,
       }
       rtps_discovery_->remove_domain_participant(config_.application_domain(), config_.application_participant_guid(), result.first->second);
       const auto pos = guid_addr_set_map_.find(result.first->second);
+      const auto prev_guid = result.first->second;
       if (pos != guid_addr_set_map_.end()) {
-        remove(proxy, result.first->second, pos, now, &relay_participant_status_reporter_);
+        remove(proxy, prev_guid, pos, now, &relay_participant_status_reporter_);
+        result.first = remote_map_.find(remote); // remote_map_ was updated by remove()
+      }
+      if (result.first != remote_map_.end()) {
```

Comment:
I think this needs an else that inserts.  Alternatively, you could combine line 42-45 with something like
```
const auto result2 = remote_map_.insert(std::make_pair(remote, src_guid));
if (!result2.second) {
  result2.first->second = src_guid;
}
```